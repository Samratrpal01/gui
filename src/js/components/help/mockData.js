export const helpProps = {
  docsVersion: undefined,
  isHosted: false,
  menderVersion: 'master',
  isEnterprise: true,
  tenantCapabilities: {},
  tokens: [],
  versions: {
    repos: {
      mender: '3.3.0',
      'mender-artifact': '3.8.0',
      'mender-binary-delta': '1.3.1',
      'mender-cli': '1.8.0',
      'mender-configure-module': '1.0.4',
      'mender-connect': '2.0.2',
      'mender-convert': '3.0.0',
      'mender-gateway': '1.0.0',
      'monitor-client': '1.2.0'
    },
    releaseDate: '01-01-2003'
  }
};
