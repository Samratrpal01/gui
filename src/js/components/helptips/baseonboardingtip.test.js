import React from 'react';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import BaseOnboardingTip, { orientations } from './baseonboardingtip';
import { defaultState, undefineds } from '../../../../tests/mockData';
import { render } from '../../../../tests/setupTests';

const mockStore = configureStore([thunk]);

describe('BaseOnboardingTip Component', () => {
  let store;
  beforeEach(() => {
    store = mockStore({ ...defaultState });
  });

  it('renders correctly', async () => {
    const { baseElement } = render(
      <Provider store={store}>
        <BaseOnboardingTip anchor={{ left: 1, top: 1, right: 1, bottom: 1 }} component={<div>Testcontent</div>} id="test" />
      </Provider>
    );
    const view = baseElement;
    expect(view).toMatchSnapshot();
    expect(view).toEqual(expect.not.stringMatching(undefineds));
  });

  it('calculates offsets depending on icon width', async () => {
    const expectedOrientation = {
      top: { top: 0, left: -15 },
      right: { top: -15, left: 15 },
      bottom: { top: 0, left: -15 },
      left: { top: -15, left: 0 }
    };
    Object.entries(orientations).forEach(([direction, orientation]) => {
      const { top, left } = orientation.offsetStyle({ top: 0, left: 0 });
      expect(top).toEqual(expectedOrientation[direction].top);
      expect(left).toEqual(expectedOrientation[direction].left);
    });
  });
});
