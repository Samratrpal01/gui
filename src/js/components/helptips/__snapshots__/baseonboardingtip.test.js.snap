// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseOnboardingTip Component renders correctly 1`] = `
.emotion-0 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-1 {
  opacity: 0.9;
}

.emotion-2 {
  z-index: 1500;
  pointer-events: auto;
}

.emotion-3 {
  background-color: #1976d2;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  color: #9e9e9e;
  font-size: 14px;
  max-width: 350px;
  padding: 12px 18px;
  width: 350px;
}

.emotion-3 a {
  color: #9e9e9e;
}

.emotion-3.MuiTooltip-tooltipPlacementTop {
  margin-left: 30px;
  margin-bottom: 0;
  margin-top: calc(30 + 12px);
}

.emotion-3.MuiTooltip-tooltipPlacementRight {
  margin-top: 15px;
}

.emotion-3.MuiTooltip-tooltipPlacementBottom {
  margin-left: 30px;
}

.emotion-3.MuiTooltip-tooltipPlacementLeft {
  margin-top: 15px;
}

.emotion-4 {
  background-color: rgba(97, 97, 97, 0.92);
  border-radius: 4px;
  color: #fff;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  padding: 4px 8px;
  font-size: 0.6875rem;
  max-width: 300px;
  margin: 2px;
  word-wrap: break-word;
  font-weight: 500;
}

.MuiTooltip-popper[data-popper-placement*="left"] .emotion-4 {
  transform-origin: right center;
  margin-right: 14px;
}

.MuiTooltip-popper[data-popper-placement*="right"] .emotion-4 {
  transform-origin: left center;
  margin-left: 14px;
}

.MuiTooltip-popper[data-popper-placement*="top"] .emotion-4 {
  transform-origin: center bottom;
  margin-bottom: 14px;
}

.MuiTooltip-popper[data-popper-placement*="bottom"] .emotion-4 {
  transform-origin: center top;
  margin-top: 14px;
}

<body>
  <div>
    <div
      class="onboard-tip"
      style="left: -14px; top: 1px; overflow: initial;"
    >
      <div
        class="tooltip onboard-icon bottom"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-0"
          data-testid="ArrowUpwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
          />
        </svg>
      </div>
    </div>
    <div
      class="emotion-1 MuiTooltip-popper MuiTooltip-popperInteractive emotion-2"
      data-popper-escaped=""
      data-popper-placement="bottom"
      data-popper-reference-hidden=""
      id="test"
      role="tooltip"
      style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
    >
      <div
        class="emotion-3 MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom emotion-4"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 133ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="content"
        >
          <div>
            Testcontent
          </div>
          <div
            class="flexbox"
          >
            <div
              style="flex-grow: 1;"
            />
            <a>
              Dismiss
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
